odl/solvers/advanced/chambolle_pock.py

```diff
@@ -0,0 +1,279 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""First-order primal-dual algorithm developed by Chambolle and Pock."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from builtins import super
+
+# External
+
+# Internal
+from odl.operator.operator import Operator
+from odl.operator.pspace_ops import ProductSpaceOperator
+
+__all__ = ('chambolle_pock_solver', 'f_cc_prox_l2_tv', 'g_prox_none')
```

Comment:
> What do you mean with changeable?

We should change it.

> The name is indeed convoluted. At least it kind of indicates that the function is related to the proximal operator of the convex conjugate of a functional related to a L2-TV optimization. But I'll try to come up with a better name.

Agree, but it is currently exposed all the way to `odl.solvers`, where users would likely find it very confusing. Naming is hard indeed, but even writing it out more explicitly could perhaps help? Also I think we need to remove the `f_`, it doesn't realy make sense in a general context. Some (out of the blue) suggestions:

`proximal_l2_conjugate_tv` Perhaps use `proximal` as a prefix for all of these, easier to look for.
`proximal_lp_conjugate_tv` We use `lp` in general, simply throw not implemented for not `l2`.
`proximal_conjugate_tv` `lp` is kindoff implicit anyway